Personality Insights sample widget

michaelkeeling opened the pull request:
* Migrating the personality insights widget from the original WEX / WDC sample code.
* Updated the code to use App Builder endpoints.  Sample proxy code is still there and also updated
* Removed the PI visualization since it was broken.  After this is merged someone should revisit this and look into creating some kind of visualization better than the basic table this widget uses.

eoehlsch commented on personality_insights/watson-personality-insights-readme.md:
```diff
@@ -0,0 +1,225 @@
+
+This simple example illustrates a basic integration. You should choose text that is relevant to a user or entity that you are modeling and you must provide enough text to generate a reliable model.  See the [Personality Insights documentation](http://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/personality-insights.html) for more guidance on text selection and recommendations.
+
+![Screen shot of "Watson Personality Insights" widget.](/personality_insights/ApplicationBuilder/presonality-insights-widget.png)
```
We should take this opportunity to correct the image filename and this link. "PRES"onality vs "PERS"onality

eoehlsch commented on personality_insights/watson-personality-insights-readme.md:
```diff
@@ -0,0 +1,225 @@
+post '/pi/model_twitter/' do
+   data = JSON.load(request.body)
+
+   url = "#{settings.personality_insights_endpoint}model_tweets/#{data["handle"]}"
```
You set "pi_endpoint" above, not "personality_insights_endpoint".

eoehlsch commented on personality_insights/watson-personality-insights-readme.md:
```diff
@@ -0,0 +1,225 @@
+post '/pi/model_text/' do
+   data = JSON.load(request.body)
+
+   url = "#{settings.personality_insights_endpoint}model_text/"
```
You set "pi_endpoint" above, not "personality_insights_endpoint".

eoehlsch commented on personality_insights/BlueMix/controllers/application_controller.rb:
```diff
@@ -0,0 +1,69 @@
+
+  
+  set :twitter_api_key, "ENTER API KEY HERE"
+  set :twitter_api_secret, "ENTER API SECRET HERE"  # you shouldn't really do this way since it's not that
```
Do you have an alternative approach in mind that you could describe briefly in a comment?

eoehlsch commented on personality_insights/ApplicationBuilder/personality-insights-twitter-widget.erb:
```diff
@@ -0,0 +1,108 @@
+  function displayModel(model) {
+    var $model_container = $(".user_model").empty(); 
+    
+    // model_container.append(JSON.stringify(model));
```
This looks like debugging that we can remove altogether instead of commenting out.

eoehlsch commented on personality_insights/ApplicationBuilder/personality-insights-twitter-widget.erb:
```diff
@@ -0,0 +1,108 @@
+    $(".go").click(function() {
+      $(".watson-status").show();
+      
+      var endpoint = "/AppBuilder/endpoint/personality_insights_twitter";
```
We should have a comment in the code here, as well as text in the readme, that suggests a new value for 'endpoint' if using the proxy.

eoehlsch commented on personality_insights/ApplicationBuilder/personality-insights-text-widget.erb:
```diff
@@ -0,0 +1,60 @@
+<%
+
+  text_to_analyze = @subject['title'].join(" ")
+model = endpoint("personality_insights_text", text: text_to_analyze)
```
We should have a comment in the code here about modifying this line to send the text to the proxy, if WEX < 11.

michaelkeeling commented:
Thanks for your review, Eric!  I've addressed all your comments and I'm going to go ahead and merge it.  I think we can address other issues -- minor or major -- in future changes sets as they come up.
